### PR TITLE
[ruby-plus-devkit] add a hab version check

### PR DESCRIPTION
**Pull request by robbkidd**
## Description

This plan must be built with Hab version 0.85.0 or greater because of its use of the `-IsPath` option when setting/pushing paths in SetupEnvironment. This change adds a version check to the Begin build callback to bail with an informative error message when the required minimum version isn't present. Otherwise, the error that appears is the slightly arcane:

```
An error occured in the build!
You cannot call a method on a null-valued expression.
At C:\hab\pkgs\core\hab-studio\0.83.0\20190712234514\bin\hab-plan-build.ps1:1447 char:5
+     $pathArray = $path.Split($separator)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
+ FullyQualifiedErrorId : InvokeMethodOnNull
```

### CODE REVIEW

File: ruby-plus-devkit/plan.ps1 [hunk 1]
```diff
@@ -42,7 +52,7 @@ function Invoke-After {
     Get-ChildItem $pkg_prefix/lib/ruby/gems/$ruby_abi_version/gems -Filter "spec" -Recurse | Remove-Item -Recurse -Force
     Get-ChildItem $pkg_prefix/lib/ruby/gems/$ruby_abi_version/cache -Recurse | Remove-Item -Recurse -Force
     Get-ChildItem $pkg_prefix -Filter "unins000.*" | Remove-Item -Recurse -Force
-    Get-ChildItem $pkg_prefix -Filter "share" | Remove-Item -Recurse -Force
```
Comment (robbkidd) [inline ruby-plus-devkit/plan.ps1:45]: Bonus trailing whitespace trim!

**Review by jeremiahsnapp — APPROVED** [review]

**Review by btm — APPROVED** [review]

**Comment by robbkidd** [comment]
I bonus'd in an update to the package metadata so that the provenance of this package is more clear when viewed on BLDR.

**Comment by robbkidd** [comment]
Tested with an older version of Hab installed. The plan appropriately bailed out early. ✅ 

```
C:\work\chef-plans [add-hab-version-check-to-plan ≡]> hab pkg build .\ruby-plus-devkit\
WARNING: Using a local Studio. To use a Docker studio, use the -D argument.
   hab-studio: Creating Studio at C:\hab\studios\work--chef-plans
» Importing origin key from standard input
≡ Imported public origin key testy-20191018184609.
» Importing origin key from standard input
≡ Imported secret origin key testy-20191018184609.
   hab-studio: Building '.\ruby-plus-devkit\' in Studio at C:\hab\studios\work--chef-plans
   : Loading C:\hab\studios\work--chef-plans\src\ruby-plus-devkit\\plan.ps1
   ruby-plus-devkit: Plan loaded
   ruby-plus-devkit: Validating plan metadata
   ruby-plus-devkit: hab-plan-build.ps1 setup
WARNING: (╯°□°）╯︵ ┻━┻ I CAN'T WORK UNDER THESE CONDITIONS!
WARNING: :habicat: I'm being built with hab 0.82.0/20190605214127. I need at least Hab 0.85.0, because I use the -IsPath option for setting/pushing paths in SetupEnvironment.
An error occured in the build!
unable to build: required minimum version of Habitat not installed
At C:\hab\studios\work--chef-plans\src\ruby-plus-devkit\plan.ps1:24 char:9
+         throw "unable to build: required minimum version of Habitat n ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (unable to build...t not installed:String) [], RuntimeException
+ FullyQualifiedErrorId : unable to build: required minimum version of Habitat not installed
```

(and in color!)

Screen Shot 2019-10-18 at 3 35 53 PM

**Comment by robbkidd** [comment]
@miah @jeremiahsnapp @btm I think the wrong-version-of-hab-bailout is working. 👍❓

(Sadly, it seems Buildkite isn't going to interpolate the `:habicat:` in the warning output, though. Apparently that only happens up in the `--- ` section "headers." For example, see the [`ruby-plus-devkit: :habicat: I think I have the version I need to build.`](https://buildkite.com/chef/chef-chef-plans-master-verify/builds/39#b5cec4d0-4d1a-493e-b290-f3b85661c1f5/142-164) (line 164) in a successful build.)

**Pull request merged**